hutch_python/utils.py

```diff
@@ -433,3 +436,22 @@ def get_fully_qualified_domain_name():
         )
         logger.debug("getfqdn failed", exc_info=True)
         return ""
+
+
+abort_msg = """
+Your RunEngine has been interrupted.  If you aborted the current run (Ctrl+C),
+the RunEngine is ready for use.  If you paused the run, you must resume,
+abort, stop, halt the RunEngine before continuing.
+"""
+
+
+class AbortSigintHandler(SignalHandler):
+    def __init__(self, RE):
+        super().__init__(signal.SIGINT, log=RE.log)
+        self.RE = RE
+
+    def handle_signals(self):
+        # Check for pause requests from keyboard.
+        if self.RE.state.is_running:
+            threading.Thread(target=self.RE.stop()).start()
```

Comment:
Does this need to be in a thread? I ask because `self.RE.stop()` is called directly here before the thread starts, the object passed into the `target` here is actually the return value from `RE.stop()`

Reply:
Good catch.  I was too lazy in mimicking the existing SIGINT handler, which actually needs the threading.

Reply:
This is one more character than yesterday's comma-at-EOL one, but in my opinion so much easier to gloss over. Very nice catch.